### ISSUE TITLE
Initializer list formatting

### ISSUE THREAD

**Issue by DanRStevens** [lairworks/nas2d-core#242]
The code base has numerous initializer lists that are formatted in a hard to read manner. The level of indentation doesn't play well across various tab size settings (2, 4, 8), and makes maintenance of alignment a pain.

I would suggest we follow a convention where multiline initializer lists place each item on their own line, with a single level increase in indentation.

Original example (scroll right for tab size 8):
```cpp
XmlDocument::XmlDocument(const std::string& documentName):	XmlNode(XmlNode::NodeType::XML_DOCUMENT),
															_errorId(0),
															_error(false)
{
```

Suggested formatting:
```cpp
XmlDocument::XmlDocument(const std::string& documentName) :
	XmlNode(XmlNode::NodeType::XML_DOCUMENT),
	_errorId(0),
	_error(false)
{
```

This issue affects the code base as a whole. I suggest we go over the code base all at once and update such formatting.

**Comment by cugone** [comment]
I like this idea, with two suggested changes: leading punctuation instead of trailing. and no leading whitespace. 

It makes reordering/adding items easier and visual studio really, really, really likes to get rid of the leading whitespace in this situation and I've resigned to not fight with it:

```cpp
XmlDocument::XmlDocument(const std::string& documentName)
: XmlNode(XmlNode::NodeType::XML_DOCUMENT)
, _errorId(0)
, _error(false)
{
```

**Comment by DanRStevens** [comment]
I view that as an improvement, though still a bit hard to read.

In terms of re-ordering items, it's better than the previous format, and about the same as my earlier suggestion. With my suggestion, there is a slight discrepancy with re-ordering due to no trailing comma on the last item. With your suggestion, there is a slight discrepancy with re-ordering due to the different punctuation for the first item.

I would find a bit of indentation would help make that easier to read, and trailing punctuation would be more familiar, and hence easier to read.

For the whitespace with Visual Studio, I assume that's the auto indent feature that fights the syntax?

**Comment by DanRStevens** [comment]
Looks like that has been closed by a combination of #330, and the changes in #345 and #346.
